### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by yakov116**
- Faster `show-associated-branch-prs-on-fork`, Request before dom ready
- Sort From
- **`isEmptyRepoRoot` does not work**
- Lint

@fregante 
> https://github.com/fregante/github-url-detection/commit/0c36c4c08636174cc039b077680ff90acb3b5d37

I followed https://github.com/sindresorhus/refined-github/pull/3505/files/92330f5ca0627ce23adb5ef216c7007008e844dc#diff-7988466746fb3fcac77295b4dd4038d1 (npm deprecate *) but that gave me an error. (I can provide a stack trace if needed 😉 )

'.blankState' does not exist on repo root. 

Direction needed.

### CODE REVIEW

**Review by yakov116 — COMMENTED** [review]
.

File: source/features/repo-age.tsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ void features.add({
 		pageDetect.isRepoRoot
 	],
 	exclude: [
-		pageDetect.isEmptyRepoRoot
+		() => !select.exists('link[rel="canonical"]')
```
Comment (yakov116) [inline source/features/repo-age.tsx:126]: I think isEmptyRepo should be isRepoHome and :arrow_up:

File: source/features/ci-link.tsx [hunk 2]
```diff
@@ -38,6 +38,9 @@ void features.add({
 	include: [
 		pageDetect.isRepo
 	],
+	exclude: [
+		() => select.exists('[aria-label="Cannot fork because repository is empty."]')
```
Comment (fregante) [inline source/features/ci-link.tsx:42]: Why this here and the rel=canonical below?

Comment (yakov116) [reply source/features/ci-link.tsx:42]: This runs on pages other then repo home. Only repo home will always has the rel.

Comment (fregante) [reply source/features/ci-link.tsx:42]: Empty repos don't have the canonical on any page

**Comment by yakov116** [comment]
Sorry let me explain slowly. 

The only time repo-age will error is on a brand new repo without any commits. 

>Empty repos don't have the canonical on any page

Correct but ci-link runs on other pages and canonical does not exists on all pages even on non-empty repos.

**Comment by fregante** [comment]
> There will only be repoHome

I see what you mean, but that's unnecessarily specific. The feature is already limited to `isRepoRoot`, which includes `isRepoHome`, and any other page that `isRepoRoo` but not `isRepoHome` already can't happen, so it's pointless to explicitly exclude it.

> the fastest way we can detect that its empty is using rel

Good thinking here. Needs a JS comment though.



> canonical does not exists on all pages even on non-empty repos

🙌

File: source/features/repo-age.tsx (hunk 1 above)
Comment (fregante) [reply source/features/repo-age.tsx:126]:
```suggestion
		// Excludes empty repos, compatible with `waitForDomReady: false`
		() => !select.exists('link[rel="canonical"]')
```

**Comment by fregante** [comment]
Can you also rename `waitForDomReady` to `awaitDomReady` everywhere?

**Comment by yakov116** [comment]
> Can you also rename `waitForDomReady` to `awaitDomReady` everywhere?

Like! Will do.

File: source/features/ci-link.tsx [hunk 3]
```diff
@@ -38,6 +38,9 @@ void features.add({
 	include: [
 		pageDetect.isRepo
 	],
-	waitForDomReady: false,
+	exclude: [
+		() => select.exists('[aria-label="Cannot fork because repository is empty."]')
```
Comment (fregante) [inline source/features/ci-link.tsx:42]: I think this can be added as `isEmptyRepo`, what do you think?

File: source/features/repo-age.tsx [hunk 4]
```diff
@@ -123,8 +123,9 @@ void features.add({
 		pageDetect.isRepoRoot
 	],
 	exclude: [
-		pageDetect.isEmptyRepoRoot
+		// Excludes empty repos, compatible with `awaitDomReady: false`
+		() => !select.exists('link[rel="canonical"]')
```
Comment (fregante) [inline source/features/repo-age.tsx:127]: I think this should be merged back as `isEmptyRepoRoot`

```js
// On empty repos, there's only isRepoHome
const isEmptyRepoRoot = () => isRepoHome() && !select.exists('link[rel="canonical"]');
```

Comment (fregante) [reply source/features/repo-age.tsx:127]: In the future, changes like this one should be made on `github-url-detection`, not here.

File: source/features/ci-link.tsx (hunk 3 above)
Comment (yakov116) [reply source/features/ci-link.tsx:42]: Yes

File: source/features/repo-age.tsx (hunk 4 above)
Comment (yakov116) [reply source/features/repo-age.tsx:127]: That was my intention. I brought it up here to explain the issue.

I will make the 2 pr's now.


In the future would you like me to create an issue there first. From my experience it usually works out best, when we discuss it here and then send the pr.

Comment (fregante) [reply source/features/repo-age.tsx:127]: Ok

**Comment by fregante** [comment]
Yes. Lint PRs shouldn’t really change the behavior, especially if like in that case it’s an improvement.

**Pull request merged**